### ISSUE TITLE
setting environment variables in command line or docker-compose has no effect

### ISSUE THREAD

**Issue by norweeg** [linuxserver/docker-heimdall#35]
I have tried again and again to change variables e.g. APP_NAME and APP_URL by setting them in the environment section of a docker-compose.yaml file (equivalent to setting them with the -e flag if starting Heimdall directly from docker), yet the /config/www/.env file that gets generated only has default settings, with none of the variables passed honored.  They variables I pass in this manner  are being set in the container, and I can echo the ones I set or display them all by running `env` in the container, but the act of setting them does not change the configuration of the application.  I have to edit the file directly to change the application's configuration.

**Comment by aptalca** [comment]
Yeah, you need to edit the .env file as described in the heimdall repo's readme.

There is nothing here that suggests you can use those environment variables in docker run parameters. The ones that are accepted by the image are specifically listed in the readme of this repo.

**Comment by norweeg** [comment]
@aptalca that is really stupid and user-unfriendly IMO.  You should be able to deploy it with a configuration, not have to edit a file after-the-fact.  Also, you yourself implied that it should be set in that manor https://github.com/linuxserver/docker-heimdall/pull/7#issuecomment-430351816

**Comment by aptalca** [comment]
?? What?

The app (heimdall) does not accept variables that way. What do you want me to do about it?

That comment had nothing to do with whether heimdall accepts it or not. I was merely commenting that if you want to set environment variables (for whatever reason), they do not have to be in the dockerfile and that you can set it in docker run.

Plus, you submitted a PR without even testing it to confirm it even does anything. I gave you the benefit of the doubt and assumed it did something.

Now you come in here and calling our work stupid? Bravo

**Comment by norweeg** [comment]
@aptalca again, you said it first.  See https://github.com/linuxserver/docker-heimdall/pull/7#issuecomment-430351816

**Comment by aptalca** [comment]
Reading comprehension dude

**Comment by norweeg** [comment]
@aptalca container best practices, dude.  All configurable parameters should be configurable by environment variabl/envfile.  Literally all other containers use it because its best practice.  Having to configure after deploy is an anti-pattern, but whatever!  have a nice life, closing all issues within minutes of opening them and making no effort to address the problem!  I shouldn't have to hack around your obstinance to deploy already-configured containers.

**Comment by aptalca** [comment]
You still fail to understand the difference between an upstream app and a wrapper container

**Comment by thelamer** [comment]
@norweeg we maintain independent software projects and docker wrapping for 100+ repos voluntarily (unpaid) and all of the build infrastructure to keep this going. But this is not an elite club, anyone can participate as long as the changes make sense and are needed by the userbase. 
Github is not reddit or 4chan, no need to take things personally. Dig into https://github.com/linuxserver/Heimdall and this repo and determine what can be changed to make an everyday users experience better. 
The best place for discussions like this and collaboration is our Discord https://discord.gg/YWrKVTn, we have a specific Heimdall room there where you can interact with other users and developers for this specific project.

**Comment by norweeg** [comment]
@aptalca you still don't understand the issue I reported.  There's nothing wrong with Heimdall, the issue is with the docker image.  There is no way, in the docker image, to deploy setting things such as APP_NAME.  You response was "duh! didn't you read the README?"  Yes, but configuring a docker container by deploying it, then editing a file and restarting the container is not how a docker container should work.  

@thelamer thank you for your response.  I reported an issue and aptalca basically closed it immediately without even considering what it was that I was reporting.  After they closed it, I pull-requested something to address it https://github.com/linuxserver/docker-heimdall/pull/36, but apparently they have no interest in improving the docker image for Heimdall.  Being able to set things like APP_URL and FORCE_HTTPS  especially important for anyone deploying this container behind a proxy because your image, as it exists today, tries to serve scripts over HTTP which browsers will refuse to run, resulting in a completely broken page.  Upon seeing that, I thought, "Oh! I forgot to set something in environment.  let me investigate...hmm... what if I set my APP_URL?" only to find that none of these options are configurable except and only by manually editing a file.  I deploy multiple heimdall conainers as part of a much larger stack using docker compose.  Differentiating them via APP_NAME, and forcing them to work properly behind a proxy by setting things like APP_URL are critical for me and the fact that they aren't already seems like an issue worth reporting.  If I can't config something through envfiles, its as good as useless and your resistance to making these options configurable at deploy via environment variables is beyond me, but since all I get is disrespect, I'll focus my efforts into hacking around your recalcitrance.  Good day!
